Compute canopy

HabibaFerchichi opened the pull request:
1- Entrer Canopy tif File
2- Add mean canopy function
3- Compute mean canopy for each CP and CE

HabibaFerchichi committed:
Entrer canopy

HabibaFerchichi committed:
Add mean canopy function

HabibaFerchichi committed:
Enter canopy tif file

erinconv commented on src/physiographic/base.py:
```diff
@@ -156,6 +156,8 @@ def _project_structure(self, file_list: str):
         self._SubBasins = os.path.join(
             self._project_path, "geographic", file_list[4])
+        self._Canopy = os.path.join(
+            self._project_path, "geographic", file_list[5])
```
The input of canopy map should be optional since not all the users will use canopy in their simulations

@HabibaFerchichi

erinconv commented on src/physiographic/base.py:
```diff
@@ -484,6 +486,9 @@ def CP_routing(self):
             self.CEfishnet, self.CPfishnet, self._DEM)
+        # Compute the mean canopy
+        self.CPfishnet, self.CEfishnet = CPfs.mean_canopy(
+            self.CEfishnet, self.CPfishnet, self._Canopy)
```
Since the canopy is optional, the function to compute the mean values should be placed outside this scope. Check on how to calculate the mean slope to have an insight

erinconv replied:
@HabibaFerchichi

erinconv commented on docs/notebooks/get_physio.ipynb:
Do not force people to use Canopy when it is intended to be an optional input file. You can clarify it within a note as was made for the Slope

erinconv commented:
@HabibaFerchichi ... Do not forget to update the parameter structure if needed.

erinconv reviewed (commented):
check my comments

HabibaFerchichi committed:
Update base.py

convert canopy input as optionnel

HabibaFerchichi committed:
Update carreauxEntiers.py

Add computeMeanCanopy function

HabibaFerchichi committed:
Update carreauxPartiels.py

Add computeMeanCanopy function in carreauxPartiels function

HabibaFerchichi committed:
Update CPfishnet.py

remove mean_canopy function from CPfishnet.py and added it in CarreauxPartiels

HabibaFerchichi committed:
Update carreauxPartiels.py

add Library rs

HabibaFerchichi committed:
Update CPfishnet.py

add mean canopy for CPfishnet and CEfishnet

HabibaFerchichi committed:
Update carreauxPartiels.py

did not work with separated function, use instead mean_canopy function in CPfishenet

HabibaFerchichi committed:
Update carreauxEntiers.py

did not work with separated function, use instead mean_canopy function in CPfishnet

HabibaFerchichi committed:
Update base.py

add canopy height as optional input

HabibaFerchichi committed:
Update base.py

Add CEfishnet Canopy in case of no file entered as input

HabibaFerchichi committed:
Update get_physio.ipynb

add note about the Canopy input

HabibaFerchichi committed:
Update carreauxPartiels.py

HabibaFerchichi committed:
Update carreauxEntiers.py

erinconv commented:
Since this PR is outdated with respect to the latest version of the code, I opened a new PR (#5) to merge this changes.
